syntax = "proto3";

package chatserver;

service ChatServer
{
    rpc LogIn (LogInRequest) returns (LogInReply) {}
    rpc LogOut (LogOutRequest) returns (LogOutReply) {}
    rpc SendMessage (stream SendMessageRequest) returns (stream SendMessageReply) {}
    rpc ReceiveMessage (ReceiveMessageRequest) returns (ReceiveMessageReply) {}
    rpc List (ListRequest) returns (ListReply) {}
    rpc Chat (stream ChatMessage) returns (stream ChatMessage) {}
}

message ChatMessage
{
    string user = 1;
    string messages = 2;
}

message LogInRequest
{
    string user = 1;
}

message LogInReply
{
    string confirmation = 1;
    string user = 2;
}

message LogOutRequest
{
    string user = 1;
}

message LogOutReply
{
    string confirmation = 1;
}

message SendMessageRequest
{
    string user = 1;
    string recipient = 2;
    string messages = 3;
}

message SendMessageReply
{
    string confirmation = 1;
    int32 status = 2;
}

message ReceiveMessageRequest
{
    string user = 1;
}

message ReceiveMessageReply
{
    string confirmation = 1;
}

message ListRequest
{
    string list = 1;
}

message ListReply
{
    string list = 1;
}
